vallie/vallie.go

```diff
@@ -0,0 +1,30 @@
+// Package vallie stands in for a stand-alone validation library
+// purely for discussion purposes.
+//
+// Assuming this is a desired approach, more fidelity will be added
+// and vallie would be extracted into it's own repo
+package vallie
+
+import (
+	"reflect"
+
```

Comment:
What is "vallie"? Its not intuitive and something that users will need to be reminded of. It sounds like a toy name. Why not just keep it on 'dig.Validate'?